spec/delayed/job_groups/plugin_spec.rb

```diff
@@ -109,6 +110,50 @@
         expect(failed_job_count).to eq 1
         expect(queued_job_count).to eq 1
         expect(job_group_count).to eq 1
+
+        # Run the last job
+        Delayed::Worker.new.work_off(1)
+        expect(failed_job_count).to eq 1
+        expect(queued_job_count).to eq 1
+        expect(job_group_count).to eq 0
+
+        # Run the completion job
+        Delayed::Worker.new.work_off(1)
+        # Completion job is invoked
+        expect(CompletionJob.invoked).to be(true)
+        expect(failed_job_count).to eq 1
+        expect(queued_job_count).to eq 0
+        expect(job_group_count).to eq 0
+      end
+
+      it "runs completion job if last job failed" do
+        Delayed::Worker.max_attempts = 1
```

Comment:
Can we set `max_attempts` to 2 and make sure we only run the completion job after the second failed attempt of `FailingJob`?

Reply:
Yep, added.